models/taskRequests.js

```diff
@@ -73,6 +73,92 @@ const fetchTaskRequestById = async (taskRequestId) => {
+      .get();
+    const [taskRequestRef] = taskRequestsSnapshot.docs;
+    const taskRequestData = taskRequestRef?.data();
+    const isCreationRequestApproved =
+      taskRequestData &&
+      taskRequestData.requestType === TASK_REQUEST_TYPE.CREATION &&
+      taskRequestData.status === TASK_REQUEST_STATUS.APPROVED;
+    if (isCreationRequestApproved) {
+      return { isCreationRequestApproved };
+    }
+    const userRequest = {
+      userId: data.userId,
+      proposedDeadline: data.proposedDeadline,
+      proposedStartDate: data.proposedStartDate,
+      description: data.description,
+      status: TASK_REQUEST_STATUS.PENDING,
+    };
+    if (!userRequest.description) delete userRequest.description;
+    if (taskRequestData) {
+      // TODO : remove after the migration of old data https://github.com/Real-Dev-Squad/website-backend/issues/1613
+      const currentRequestors = taskRequestData.requestors;
+      let alreadyRequesting = currentRequestors.some((requestor) => requestor === data.userId);
+      // End of old logic
+      const currentRequestingUsers = taskRequestData.users;
+      alreadyRequesting = currentRequestingUsers.some((requestor) => requestor.userId === data.userId);
+      if (alreadyRequesting) {
+        return { alreadyRequesting };
+      }
+      // TODO : remove after the migration of old data https://github.com/Real-Dev-Squad/website-backend/issues/1613
+      const updatedRequestors = [...currentRequestors, data.userId];
+      // End of old logic
+      const updatedUsers = [...currentRequestingUsers, userRequest];
+      const updatedTaskRequest = {
+        requestors: updatedRequestors,
+        users: updatedUsers,
+        lastModifiedBy: authenticatedUsername,
+        lastModifiedAt: Date.now(),
+      };
+      await taskRequestsCollection.doc(taskRequestRef.id).update(updatedTaskRequest);
+      return {
+        id: taskRequestRef.id,
+        isCreate: false,
+        taskRequest: {
+          ...taskRequestData,
+          ...updatedTaskRequest,
+        },
+      };
+    }
+    const newTaskRequest = {
+      requestors: [data.userId],
+      status: TASK_REQUEST_STATUS.PENDING,
+      taskTitle: data.taskTitle,
+      taskId: data.taskId,
+      externalIssueUrl: data.externalIssueUrl,
+      requestType: data.requestType,
+      users: [userRequest],
+      createdBy: authenticatedUsername,
+      createdAt: Date.now(),
+      lastModifiedBy: authenticatedUsername,
+      lastModifiedAt: Date.now(),
+    };
+    if (!newTaskRequest.externalIssueUrl) delete newTaskRequest.externalIssueUrl;
+    if (!newTaskRequest.taskId) delete newTaskRequest.taskId;
+    if (!newTaskRequest.taskTitle) delete newTaskRequest.taskTitle;
+    const newTaskRequestRef = await taskRequestsCollection.add(newTaskRequest);
+    return {
+      isCreate: true,
+      taskRequest: newTaskRequest,
+      id: newTaskRequestRef.id,
+    };
+  } catch (err) {
+    logger.error("Error creating a task request", err);
```

Comment:
While you're using error handling with `try...catch` blocks, and logging errors using `logger.error`, you're not providing meaningful error messages in your error responses. When an error occurs and you return a response using `res.boom`, you should include more detailed error messages. 
This will help us debug and make it easier to understand what exactly went wrong

Reply:
All the known cases are handled and returns a meaningful messages. 
Usually API response messages are written considering the consumer of the API. Not for debugging purposes. For debugging we use server logs.